EIPS/eip-223.md

```diff
@@ -0,0 +1,369 @@
+---
+eip: 223
+title: 223 Token with communication model
+description: Token with event handling and communication model
+author: Dexaran (@Dexaran)
+discussions-to: https://ethereum-magicians.org/t/erc-223-token-standard/12894
+status: Review
+type: Standards Track
+category: ERC
+created: 2017-05-03
+---
+
+## Abstract
+
+The following describes an interface for fungible tokens that supports a `tokenReceived` callback to notify contract recipients when tokens are received.
+
+## Motivation
+
+This token introduces a communication model for contracts that can be utilized to straighten the behavior of contracts that interact with such tokens. Specifically, this proposal:
+
+1. Informs receiving contracts of tokens, as opposed to [ERC-20](./eip-20.md) where the recipient of a token transfer gets no notification.
+2. Is more gas-efficient when depositing tokens to contracts.
+3. Allows for `_data` recording for financial transfers.
+
+
+## Specification
+
+
+### Methods
+
+Contracts intending to receive these tokens MUST implement `tokenReceived`.
+
+Tokens transferred to contracts not implementing `tokenReceived` as described below MUST revert.
+
+
+#### `totalSupply`
+
+```solidity
+function totalSupply() view returns (uint256 totalSupply)
+```
+
+Get the total token supply
+
+#### `name`
+
+```solidity
+function name() view returns (string _name)
+```
+
+Get the name of token
+
+#### `symbol`
+
+```solidity
+function symbol() view returns (bytes32 _symbol)
+```
+
+Get the symbol of token
+
+#### `decimals`
+
+```solidity
+function decimals() view returns (uint8 _decimals)
+```
+
+Get decimals of token
+
+#### `standard`
+
+```solidity
+function standard() view returns (string _standard)
+```
+
+Get the standard of token contract. For some services it is important to know how to treat this particular token. If token supports [ERC-223](./eip-223.md) standard then it must explicitly tell that it does.
+
+This function **MUST** return "ERC-223" for this token standard. If no "standard()" function is implemented in the contract then the contract must be considered to be ERC-20.
```

Comment:
Use EIP-165 or some other introspection EIP. If you really want this function, make it its own EIP. But this function is outside the scope of this EIP.

Reply:
Ok, I can make it into separate EIP. 

However in some cases its necessary to recognize "if the token is ERC-223 or something else" and it could be an easy way. If it returns `erc223` then its ERC-223, otherwise "something else".

Reply:
That's why I suggested using EIP-165 instead.